stan-dev/math
# Bugfix/415 cholesky decompose

#### Submisison Checklist


#### Summary:
- Fixes #415. Fixes the implementation of `cholesky_decompose` so the upper-triangular elements have a `vari` on the autodiff stack.
- Adds tests to check that functions that return `stan::math::var` have `vari`s that are on the autodiff stack.

#### Intended Effect:
Fixes bug in implementation.

#### How to Verify:
To verify that `cholesky_decompose` is fixed, run:
```
./runTests.py test/unit/math/rev/mat/fun/cholesky_decompose_test.cpp
```

Run the rest of the changed tests.

#### Side Effects:
Fixes a bug.

#### Documentation:
None.

#### Reviewer Suggestions: 
@rtrangucci 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)
